dfsclient/src/main/java/org/hpccsystems/dfs/client/BinaryRecordReader.java

```diff
@@ -914,6 +940,59 @@ private BigDecimal getSignedDecimal(int numDigits, int precision, int dataLen) t
         return ret;
     }
 
+    private void trimStringInScratchBuffer(boolean isUnicode, int[] range)
```

Comment:
let's add element level comments, as discussed, see if we can use Character.isWhitespace